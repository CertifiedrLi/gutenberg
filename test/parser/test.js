/**
 * Node dependencies
 */
import fs from 'fs';
import path from 'path';
import { exec } from 'child_process';

/**
 * External dependencies
 */
import rimraf from 'rimraf';
import { uniq } from 'lodash';

const parserBinary = process.env.BINARY;
const fixturesDir = path.join( __dirname, 'fixtures' );
const tempDir = path.join( __dirname, 'temp' );

// We expect 2 different types of files for each fixture:
//  - fixture.html            : original content
//  - fixture.parsed.json     : parser output
const fileBasenames = uniq(
	fs.readdirSync( fixturesDir )
		.filter( f => /(\.html|\.json)$/.test( f ) )
		.map( f => f.replace( /\..+$/, '' ) )
);

describe( 'parser', () => {
	beforeAll( () => {
		rimraf.sync( tempDir );
		fs.mkdirSync( tempDir );
	} );

	afterEach( () => {
		rimraf.sync( tempDir );
	} );

	fileBasenames.forEach( f => {
		it( f, ( done ) => {
			const inputFile = path.resolve( fixturesDir, f + '.html' );
			const temporaryFileOutput = path.resolve( tempDir, f + '.json' );

			exec( `${ parserBinary } ${ inputFile } ${ temporaryFileOutput }`, ( err ) => {
				if ( err ) {
					throw new Error(
						'Parsing error for fixture file: ' + f
					);
				}

				const parserOutput = fs.readFileSync( temporaryFileOutput, 'utf8' );
				expect( JSON.parse( parserOutput ) ).toMatchSnapshot();
				done();
			} );
		} );
	} );
} );
