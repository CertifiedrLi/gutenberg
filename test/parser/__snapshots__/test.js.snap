// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`parser autoclosing-block 1`] = `
Array [
  Object {
    "attrs": Object {
      "ref": 123,
    },
    "blockName": "core/block",
    "innerBlocks": Array [],
    "innerHTML": "",
  },
  Object {
    "attrs": Object {},
    "innerHTML": "
",
  },
]
`;
